File: chat2db-community-server/chat2db-community-query-excel/query-excel-domain-api/src/main/java/ai/chat2db/community/query/excel/domain/api/model/SavedQueryView.java
package ai.chat2db.community.query.excel.domain.api.model;

import java.util.Date;
import java.util.List;

import lombok.Data;

@Data
public class SavedQueryView {

    private Long id;

    private Long workspaceId;

    private Long datasetId;

    private Integer datasetVersion;

    private String name;

    private String description;

    private List<ViewDimension> dimensions;

    private List<ViewMeasure> measures;

    private List<String> rowFields;

    private List<String> columnFields;

    private List<ViewFilter> filters;

    private List<ViewSort> sort;

    private Integer pageSize;

    private String status;

    private Integer version;

    private Long ownerId;

    private Date gmtCreate;

    private Date gmtModified;
}

File: chat2db-community-client/src/typings/savedQueryView.ts
export interface ViewDimension {
  fieldId?: string;
  role?: string;
  sortDirection?: string;
}

export interface ViewMeasure {
  fieldId?: string;
  aggregation?: string;
}

export interface ViewFilter {
  fieldId?: string;
  filterType?: string;
  operator?: string;
  value?: string;
  values?: string[];
}

export interface ViewSort {
  fieldId?: string;
  direction?: string;
}

export interface SavedQueryView {
  id?: number;
  workspaceId?: number;
  datasetId?: number;
  datasetVersion?: number;
  name?: string;
  description?: string;
  dimensions?: ViewDimension[];
  measures?: ViewMeasure[];
  rowFields?: string[];
  columnFields?: string[];
  filters?: ViewFilter[];
  sort?: ViewSort[];
  pageSize?: number;
  status?: string;
  version?: number;
  ownerId?: number;
  gmtCreate?: string;
  gmtModified?: string;
}

export interface SavedQueryViewListParams {
  workspaceId?: number;
  searchKey?: string;
  pageNo: number;
  pageSize: number;
}

export interface PreviewSavedQueryViewParams {
  id: number;
  pageNo?: number;
  pageSize?: number;
  filterOverrides?: string;
}